libs/ui/design-system/src/lib/input.tsx

```diff
@@ -1,28 +1,66 @@
-import type * as React from 'react'
-
-import { Input as InputPrimitive } from '@nmit-coursition/ui/primitives/input'
+import * as InputPrimitive from '@nmit-coursition/ui/primitives/input'
 import { Label } from '@nmit-coursition/ui/primitives/label'
 import { cn } from '@nmit-coursition/ui/utils'
+import { type VariantProps, cva } from 'class-variance-authority'
+
+type OverrideProps<T, V> = V & Omit<T, keyof V>
 
-type InputType = React.ComponentProps<'input'>['type']
+const inputVariants = cva('', {
+  variants: {
+    variant: {
+      primary: '',
+      valid: '', // green borders
+      inValid: '',
+      disabled: '',
+    },
+  },
+})
 
-interface InputProps extends React.ComponentProps<'input'> {
+type Orientation = 'horizontal' | 'vertical'
+
+interface InputProps
+  extends OverrideProps<
+      InputPrimitive.RootProps,
+      { type: InputPrimitive.RootProps['type']; placeholder: string; id: string }
+    >,
+    VariantProps<typeof inputVariants> {
   label: string
-  placeholder: string
-  type: InputType
-  id: string
   subtext?: string
   disabled?: boolean
-
-  // todo: use mix props
+  orientation?: Orientation
   containerClassName?: string
 }
 
-export function Input({ label, placeholder, type, id, subtext, disabled, containerClassName, ...rest }: InputProps) {
+export function Input({
+  label,
+  placeholder,
+  type,
+  id,
+  subtext,
+  disabled,
+  variant,
+  orientation = 'vertical',
+  containerClassName,
+  ...rest
+}: InputProps) {
   return (
-    <div className={cn('grid w-full items-center gap-1.5', containerClassName)}>
+    <div
+      className={cn(
+        'flex w-full items-center justify-center gap-1.5',
+        orientation == 'vertical' ? 'flex-col items-start' : '',
```

Comment:
This is what `variants` are for. No need to manually do this switch.

Reply:
this would mean that we need the variant to be applied to the div, and not the primitive per se :/

Reply:
@BleedingDev wdyt?